src/db.c

```diff
@@ -2559,8 +2566,9 @@ void db_col_transaction_commit(struct db_filter_col *col)
 	if (snap->shadow) {
 		/* leave the shadow intact, but drop the next snapshot */
 		if (snap->next) {
-			snap->next = snap->next->next;
-			_db_snap_release(snap->next);
+			struct db_filter_snap *tmp = snap->next;
+			snap->next = tmp->next;
+			_db_snap_release(tmp);
 		}
 		return;
 	}
```

Comment:
(This comment is for a couple lines down, but Github won't let me comment there.)

Since we had issues handling aborts, I wrote a quick test for this large block of code where the col's filter count didn't match the snap's filter count.

Here's the test I threw together:
https://github.com/drakenclimber/libseccomp/commit/38147745712293fe4fee21c87911fae90a9531b1

And here's the code coverage (prior to this test these lines weren't being hit):
https://coveralls.io/builds/62918086/source?filename=src%2Fdb.c#L2589

Interestingly this leads to a valgrind failure, but I haven't looked into it yet.

Reply:
Thanks for the test!  I've included it in my dev branch (which I'm going to refresh here in a few minutes) and used it to fix a number of bugs.